### ISSUE TITLE
update_packages(force = TRUE) does not reinstall package

### ISSUE THREAD

**Issue by dpprdan** [r-lib/remotes#521]
`update_packages(force = TRUE)` does not reinstall a package, which it should, if I understand the documentation correctly. 

> Force installation, even if the remote state has not changed since the previous install.

It seems to me that the `force` argument in `update.package_deps()` is not evalutated for packages where `object$diff == CURRENT`?

``` r
library(remotes)
update_packages("lookup", force = TRUE)
install_github("jimhester/lookup")
#> Using github PAT from envvar GITHUB_PAT
#> Skipping install of 'lookup' from a github remote, the SHA1 (eba63db4) has not changed since last install.
#>   Use `force = TRUE` to force installation
```

<sup>Created on 2020-07-15 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

<details>

<summary>

Session info

</summary>

``` r
devtools::session_info()
#> - Session info ---------------------------------------------------------------
#>  setting  value                       
#>  version  R version 4.0.2 (2020-06-22)
#>  os       Windows 10 x64              
#>  system   x86_64, mingw32             
#>  ui       RTerm                       
#>  language en                          
#>  collate  German_Germany.1252         
#>  ctype    German_Germany.1252         
#>  tz       Europe/Berlin               
#>  date     2020-07-15                  
#> 
#> - Packages -------------------------------------------------------------------
#>  package     * version    date       lib source                        
#>  assertthat    0.2.1      2019-03-21 [1] CRAN (R 4.0.2)                
#>  backports     1.1.7      2020-05-13 [1] CRAN (R 4.0.0)                
#>  callr         3.4.3      2020-03-28 [1] CRAN (R 4.0.2)                
#>  cli           2.0.2      2020-02-28 [1] CRAN (R 4.0.2)                
#>  crayon        1.3.4      2017-09-16 [1] CRAN (R 4.0.2)                
#>  curl          4.3        2019-12-02 [1] CRAN (R 4.0.2)                
#>  desc          1.2.0      2018-05-01 [1] CRAN (R 4.0.2)                
#>  devtools      2.3.0      2020-04-10 [1] CRAN (R 4.0.2)                
#>  digest        0.6.25     2020-02-23 [1] CRAN (R 4.0.2)                
#>  ellipsis      0.3.1      2020-05-15 [1] CRAN (R 4.0.2)                
#>  evaluate      0.14       2019-05-28 [1] CRAN (R 4.0.2)                
#>  fansi         0.4.1      2020-01-08 [1] CRAN (R 4.0.2)                
#>  fs            1.4.2      2020-06-30 [1] CRAN (R 4.0.2)                
#>  glue          1.4.1      2020-05-13 [1] CRAN (R 4.0.2)                
#>  highr         0.8        2019-03-20 [1] CRAN (R 4.0.2)                
#>  htmltools     0.5.0      2020-06-16 [1] CRAN (R 4.0.2)                
#>  knitr         1.29       2020-06-23 [1] CRAN (R 4.0.2)                
#>  magrittr      1.5        2014-11-22 [1] CRAN (R 4.0.2)                
#>  memoise       1.1.0      2017-04-21 [1] CRAN (R 4.0.2)                
#>  pkgbuild      1.1.0      2020-07-13 [1] CRAN (R 4.0.2)                
#>  pkgload       1.1.0      2020-05-29 [1] CRAN (R 4.0.2)                
#>  prettyunits   1.1.1      2020-01-24 [1] CRAN (R 4.0.2)                
#>  processx      3.4.3      2020-07-05 [1] CRAN (R 4.0.2)                
#>  ps            1.3.3      2020-05-08 [1] CRAN (R 4.0.2)                
#>  R6            2.4.1      2019-11-12 [1] CRAN (R 4.0.2)                
#>  remotes     * 2.1.1.9001 2020-07-15 [1] Github (r-lib/remotes@76c05ce)
#>  rlang         0.4.6      2020-05-02 [1] CRAN (R 4.0.2)                
#>  rmarkdown     2.3        2020-06-18 [1] CRAN (R 4.0.2)                
#>  rprojroot     1.3-2      2018-01-03 [1] CRAN (R 4.0.2)                
#>  sessioninfo   1.1.1      2018-11-05 [1] CRAN (R 4.0.2)                
#>  stringi       1.4.6      2020-02-17 [1] CRAN (R 4.0.0)                
#>  stringr       1.4.0      2019-02-10 [1] CRAN (R 4.0.2)                
#>  testthat      2.3.2      2020-03-02 [1] CRAN (R 4.0.2)                
#>  usethis       1.6.1      2020-04-29 [1] CRAN (R 4.0.2)                
#>  withr         2.2.0      2020-04-20 [1] CRAN (R 4.0.2)                
#>  xfun          0.15       2020-06-21 [1] CRAN (R 4.0.2)                
#>  yaml          2.2.1      2020-02-01 [1] CRAN (R 4.0.0)                
#> 
#> [1] C:/Users/daniel/Documents/.R/win-library
#> [2] C:/Program Files/R/R-4.0.2/library
```

</details>

**Comment by dpprdan** [comment]
This affects both CRAN and non-CRAN packages.

**Comment by jimhester** [comment]
Closed by e09a5dc4a4bf68f1d9fdbad2e0ff088c8d8acedc, the force argument doesn't really make sense for `update_packages()`.

**Comment by dpprdan** [comment]
> the force argument doesn't really make sense for `update_packages()`.

@jimhester I've found it quite useful (in theory at least) to reinstall a bunch of packages without having to know which remote (CRAN, GitHub, GitLab, ...) or repo ("is {lookup} from 'jimhester' or 'r-lib'?) they are from. Is there another way to do this easily?
